File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<int> s;

    void dfs(vector<int>& nums,int i){
        if(i==nums.size()){
            res.push_back(vector<int>(s));
            return;
        }
        s.push_back(nums[i]);
        dfs(nums,i+1);
        s.pop_back();
        dfs(nums,i+1);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        dfs(nums,0);
        return res;
    }
};

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        s =[]
        def dfs(i):
            if i ==len(nums):
                res.append(s.copy())
                return 
            s.append(nums[i])
            dfs(i+1)
            s.pop()
            dfs(i+1)
        dfs(0)
        return res